src/main/resources/avro/alleleAnnotationmethods.avdl

```diff
@@ -0,0 +1,244 @@
+@namespace("org.ga4gh.methods")
+protocol AlleleAnnotationMethods {
+
+/*This protocol defines methods use to mine pre-calculated annotation sets.*/
+
+import idl "methods.avdl";
+import idl "variants.avdl";
+import idl "alleleAnnotations.avdl";
+import idl "sequenceAnnotations.avdl";
+import idl "ontologies.avdl";
+
+
+
+/******************  /variantannotations/search  *********************/
+/** This request maps to the body of `POST /variantannotations/search` as JSON. */
+record SearchVariantAnnotationsRequest {
+  /** Required. The IDs of the annotation sets to search over. */
+  array<string> annotationSetIds = [];
+
+  /**
+  Only return annotation for variants which have exactly this name
+  (case-sensitive, exact match).
+  */
+  union { null, string } name = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this name.
+  One of this field or `referenceId` or `features` is required.
+  (case-sensitive, exact match)
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this ID. One
+  of this field or `referenceName` or `features` is required.
+  */
+  union { null, string } referenceId = null;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The beginning of the window (0-based, inclusive) for which variants with
+  overlapping reference alleles should be returned.
+  Genomic positions are non-negative integers less than reference length.
+  Requests spanning the join of circular genomes are represented as
+  two requests one on each side of the join (position 0).
+  */
+  long start;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The end of the window (0-based, exclusive) for which variants with
+  overlapping reference alleles should be returned.
+  */
+  long end;
+
+
+  /**
+  Only return variant annotations for any of these features
+  If null, return all variant annotations in specified window.
+  */
+  union { null, array<org.ga4gh.models.Feature> } features;
```

Comment:
How does the matching work? Is this matching feature IDs of TranscriptEffects or is this doing a live JOIN against the regions annotated by those features?

If it's the former, this should be named more specifically and should refer to the feature_id. If it's the latter, I'm not sure what this buys you beyond the above start/end parameters.